README.md

```diff
@@ -11,6 +11,65 @@ part of the array each recursive call considers.
 
 ## Runtime Analysis
 
-Analyse the time complexity of your implementation and give a $\Theta$ bound for
+Analyze the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+Complexity Analysis: 
+Here is my code for the activity. 
+function mergesort(array) {
+    let n = array.length;
+
+    // Start the subarrays of 1 and double with each iteration 
+    for(let length = 1; length < n; length *= 2){
+        for(let left = 0; left < n - 1; left += 2 * length) {
+            let mid = Math.min(left + length - 1, n - 1); 
+            let right = Math.min(left + 2 * length - 1, n-1); 
+            merge(array, left, mid, right);
+        }
+    }
+    
+    return array;
+}
+
+function merge(array, left, mid, right) {
+    let start = left; 
+    let end = mid + 1; 
+
+    while (start <= mid && end <= right) {
+        if (array[start] <= array[end]) {
+            start++; 
+        } else {
+            let value = array[end]; 
+            let index = end; 
+
+            while (index !== start) {
+                array[index] = array[index - 1]; 
+                index--; 
+            }
+
+            array[start] = value; 
+
+            start++;
+            mid++;
+            end++;
+        }
+    }
+}
+
+Looking back at my outer loop function the length doubles with every iteration done. This will execute log_2n times in the runtime analysis. With that execution 
+it will create _O_(log n) for all the iterations done in it. 
+
+The inner loop divides the arrays into subarrays depending on their length, making this iteration _O_(n (2 x length)). This happens twice with the left and right. 
+This will just make _O_(n^2)
```

Comment:
Not following you here. The inner loop runs at most for n iterations (when `length` is 1), so it can't be $n^2$ altogether.

Reply:
So it is theta (n log n). I should have trusted my gut instead of what my friend was telling me.

Reply:
for the worse case scenario.

Reply:
Do I need to change the readme file or no?

Reply:
So the same as the "normal" mergesort? Aren't you doing more work?

Reply:
I am doing more work on this. The only reason it would theta (n log n) is if I had some sort of array that was temporary. Since there is shifting involved in the code. I have to have _O_ (n) which I would have to merge that will _O_( n log n). Merging the two would make _O_(n x n log n) = _O_ (n^2 log n). That was my thought process through it. I guess I did not explain it that well.

Reply:
the _O_ (n) comes from the merge function and the standard time for merge sort is _O_(n log n)

Reply:
That sounds like it's going in the right direction.